### ISSUE TITLE
Since there is no grouping of option yet, can we insert arbitrary html in the name?

### ISSUE THREAD

**Issue by kevinelliott** [thisbeyond/solid-select#34]
I'd like to have a TAG in front of the name of the results to indicate what kind of content it is, since there's no grouping yet. Is this possible?

**Comment by martinpengellyphillips** [comment]
Yes, you can put arbitrary tags in the format option to control what's rendered. I'll try to add an example in the coming days (along with how groups can be emulated using disabled options and some custom styling).

**Comment by martinpengellyphillips** [comment]
FYI I've added a grouping example to the demo site now - search for 'Grouping' in the select box. 
The main issue is that the group headers are included in the navigation - I think I will add something to the core to deal with that longer term (and possibly bundle the `createGroupedOptions` approach once tested more.

**Comment by kevinelliott** [comment]
Thanks, how can I use this with async?

**Comment by kevinelliott** [comment]
<img width="725" alt="Screenshot 2023-03-07 at 7 45 03 PM" src="https://user-images.githubusercontent.com/123112/223614372-4b27db55-2b14-4df8-8d98-bcc1e378fe47.png">

**Comment by kevinelliott** [comment]
@martinpengellyphillips Any thoughts?

**Comment by martinpengellyphillips** [comment]
Did you try modifying the example to use `createAsyncOptions` helper?
